packages/yarnpkg-core/sources/Project.ts

```diff
@@ -884,6 +912,26 @@ export class Project {
       }
     });
 
+    const addedPackages: Array<string> = [];
+    const removedPackages: Array<string> = [];
+
+    for (const [locatorHash, pkg] of originalPackages)
+      if (!initialPackages.has(locatorHash) && !this.tryWorkspaceByLocator(pkg))
+        addedPackages.push(structUtils.prettyLocator(this.configuration, pkg));
+
+    for (const [locatorHash, pkg] of initialPackages)
+      if (!originalPackages.has(locatorHash) && !this.tryWorkspaceByLocator(pkg))
+        removedPackages.push(structUtils.prettyLocator(this.configuration, pkg));
+
+    addedPackages.sort();
+    removedPackages.sort();
+
+    if (addedPackages.length > 0)
+      opts.report.reportInfo(MessageName.RESOLUTION_NOT_CACHED, `${formatUtils.pretty(this.configuration, `+`, formatUtils.Type.ADDED)} ${addedPackages.join(`, `)}`);
+
+    if (removedPackages.length > 0)
+      opts.report.reportInfo(MessageName.RESOLUTION_NOT_CACHED, `${formatUtils.pretty(this.configuration, `-`, formatUtils.Type.REMOVED)} ${removedPackages.join(`, `)}`);
```

Comment:
I think the display could be improved.

This is what it looks like with `preferTruncatedLines` (I feel like it's a bit too compact):
![image](https://github.com/yarnpkg/berry/assets/32596136/8bc133d5-8b57-4177-b244-a4562762a2ad)

And this is what it looks like without (too much information):
![image](https://github.com/yarnpkg/berry/assets/32596136/219ad840-7200-4941-80ab-04edb42d8549)

Reply:
Agree; I fixed it (the amount of items printed in the line depends on the terminal size):

<img width="784" alt="Screenshot 2023-06-21 at 11 59 08" src="https://github.com/yarnpkg/berry/assets/1037931/f9ac0db0-2427-4f43-8e21-66bad222912b">